docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,598 @@
+
+The BGP feature in ovn-kubernetes only disables SNAT for traffic that is
+destined to external networks, not for traffic that is destined to other pods in
+the cluster.
+
+In no-overlay mode, we want to fully leverage BGP and avoid the SNAT step on the
+gateway router for intra-cluster traffic as well. Pod-to-pod traffic within the
+same network will hit the BGP-learned routes to reach its destination node,
+without being encapsulated in a Geneve tunnel and without being SNATed to the
+node IP.
+
+The only exception is for pod-to-other_node traffic, to ensure nodePort services
+can be accessed across networks. The egress traffic will be SNATed to the node
+IP before leaving the node.
+
+#### Isolation between UDNs with no-overlay mode enabled
+
+The [existing UDN isolation
+mechanism](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5186/commits/8f6e7d30ee5f4926a21e2de75488aad80344814b)
+will still be in place. Pods in different UDNs will be isolated from each other
+and from the default network by means of the existing OVN ACLs on every node
+logical switch: ACLs are enforced on outgoing traffic at each advertised UDN
+switch, verifying whether both source and destination IPs belong to the same
+advertised UDN. If the destination IP does not come from the same UDN subnet as
+the source IP, traffic is dropped. The no-overlay mode will not affect the
+isolation between UDNs or between UDNs and the default network.
+
+### Workflow
+
+#### Enable No-overlay Mode for the Default Network
+
+No-overlay mode for the default network must be enabled before OVN-Kubernetes
+pods start. Therefore, the following configuration manifests shall be created
+before deploying the OVN-Kubernetes pods.
+
+1. The cluster admin enables no-overlay mode by creating the well-known
+   `default` UDN CR with `spec.layer3.encapsulation` set to `None`.
+
+    ```yaml
+    apiVersion: k8s.ovn.org/v1
+    kind: ClusterUserDefinedNetwork
+    metadata:
+      # 'default' is the reserved name to represent the cluster default network
+      name: default
+    spec:
+      # use an empty selector to select all namespaces
+      namespaceSelector: {}
+      network:
+        topology: Layer3
+        layer3:
+          role: Primary
+          mtu: 1500
+          subnets:
+          - cidr: 10.244.0.0/16
+            hostSubnet: 24
+          encapsulation: None
+    ```
+
+1. OVN-kubernetes cluster manager watches the `default` CUDN CR. If
+   `spec.layer3.encapsulation` equals `None`, cluster manager will create a
+   `RouteAdvertisement` CR which will advertise pod subnets in the default
+   network:
+
+    ```yaml
+    apiVersion: k8s.ovn.org/v1
+    kind: RouteAdvertisements
+    metadata:
+      name: default
+    spec:
+      # nodeSelector must be empty, since we don't support a network in a overlay and no-overlay hybrid mode.
+      nodeSelector: {}
+      frrConfigurationSelector:
+        matchLabels:
+          network: default
```

Comment:
Missing details on the reasoning on what FRRConfiguration should be selected. Why match labels `network: default`?

Reply:
The no-overlay mode relies on BGP routers to exchange the routes to pod subnets. We use the selected FRRConfiguration to connect to the remote BGP router. The label is just an example.